src/components/tests/CharacterItem.test.tsx

```diff
@@ -0,0 +1,54 @@
+import { render, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { ModifiedCharacter } from '../../api/starWars-types'
+import CharacterItem from '../CharacterItem'
+
+const withBmiCharacter: ModifiedCharacter = {
+  id: 1,
+  name: 'Full Name',
+  image: 'image.com',
+  gender: 'gender',
+  species: 'species',
+  bmi: 24.68,
+  height: 1.72,
+  mass: 73,
+}
+
+const withoutBmiCharacter: ModifiedCharacter = {
+  id: 1,
+  name: 'Full Name',
+  image: 'image.com',
+  gender: 'gender',
+  species: 'species',
+}
+
+describe('CharacterItem', () => {
+  test('renders a character item with bmi if they have', () => {
+    render(<CharacterItem character={withBmiCharacter} />)
+    const character = screen.getByText(withBmiCharacter.name)
+    expect(character).toBeInTheDocument()
+    const bmi = screen.getByText(/bmi/i)
+    expect(bmi).toBeInTheDocument()
+  })
+
+  test('renders a character item witout bmi if not', () => {
+    render(<CharacterItem character={withoutBmiCharacter} />)
+    const bmi = screen.queryByText(/bmi/i)
+    expect(bmi).not.toBeInTheDocument()
+  })
+
+  test('show details when users click the button and hide them when users click it again', async () => {
```

Comment:
Ah ic I thought it was a bug that `View details` button remains even the hidden items were shown

Reply:
![2022-12-18 21 57 53](https://user-images.githubusercontent.com/44686790/208364356-600df8e4-1110-4498-aa11-5a5baf18d08f.gif)